_source/_plugins/gist_tag.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+# A tag for Gist include
+# Based on https://github.com/jekyll/jekyll-gist
+require "cgi"
+require "uri"
+require "net/http"
+require "octokit"
+
+Net::OpenTimeout = Class.new(RuntimeError) unless Net.const_defined?(:OpenTimeout)
+Net::ReadTimeout = Class.new(RuntimeError) unless Net.const_defined?(:ReadTimeout)
+
+module Jekyll
+  module Gist
+    class GistTag < Liquid::Tag
+      def self.client
+        @client ||= Octokit::Client.new :access_token => ENV["JEKYLL_GITHUB_TOKEN"]
+      end
+
+      def render(context)
+        @encoding = context.registers[:site].config["encoding"] || "utf-8"
+        @settings = context.registers[:site].config["gist"]
+        if (tag_contents = determine_arguments(@markup.strip))
+          gist_id  = get_gist_id(tag_contents[0])
+          filename = tag_contents[1]
+          gist_id  = context[gist_id]  if context.key?(gist_id)
+          filename = context[filename] if context.key?(filename)
+
+          noscript_tag = gist_noscript_tag(gist_id, filename)
+          script_tag   = gist_script_tag(gist_id, filename)
+
+          "#{noscript_tag}#{script_tag}"
+        else
+          raise ArgumentError, <<~ERROR
+            Syntax error in tag 'gist' while parsing the following markup:
+              '{% #{raw.strip} %}'
+            Valid syntax:
+              {% gist user/1234567 %}
+              {% gist user/1234567 foo.js %}
+              {% gist 28949e1d5ee2273f9fd3 %}
+              {% gist 28949e1d5ee2273f9fd3 best.md %}
+          ERROR
+        end
+      end
+
+      private
+
+      def get_gist_id(url)
+        URI(url).path.split('/').last
+      end
+
+      def determine_arguments(input)
+        matched = input.match(%r!\A([\S]+|.*(?=\/).+)\s?(\S*)\Z!)
+        [matched[1].strip, matched[2].strip] if matched && matched.length >= 3
+      end
+
+      def gist_script_tag(gist_id, filename = nil)
+        url = "https://gist.github.com/#{gist_id}.js"
+        url = "#{url}?file=#{filename}" unless filename.to_s.empty?
+
+        "<script src=\"#{url}\"> </script>"
```

Comment:
We haven't used any embedded Gists in the past, so this might need to be adjusted to have a div that adds the appropriate spacing above/below.

Reply:
updated to use similar div as others